client/client.go

```diff
@@ -68,6 +69,10 @@ func (c *Client) DoAndUnmarshal(ctx context.Context, r *Request, v interface{})
 		return err
 	}
 
+	if response.Close {
+		defer response.Body.Close()
+	}
+
 	if response.StatusCode == http.StatusNoContent || response.ContentLength == 0 {
 		return nil
```

Comment:
We should discard the stream and close the reader in this case I guess?

Reply:
Yes, but you need to read and throw away the reader (by throwing it in io.Discard) for the connection pooling in the internal Go HTTP client to work properly.